serde/src/de/impls.rs

```diff
@@ -70,6 +103,32 @@ impl Deserialize for () {
 
 ///////////////////////////////////////////////////////////////////////////////
 
+#[cfg(all(not(feature = "std"), not(feature = "rustc_unicode")))]
```

Comment:
Is this not available in one of the Unicode modules? If not, can you pull this Unicode stuff into a separate private module?